stream-transformation-test/stream-transformation-it/src/test/java/uk/gov/justice/framework/tools/replay/StreamTransformationIT.java

```diff
@@ -0,0 +1,210 @@
+import static uk.gov.justice.services.test.utils.core.messaging.JsonEnvelopeBuilder.envelope;
+import static uk.gov.justice.services.test.utils.core.messaging.MetadataBuilderFactory.metadataWithRandomUUID;
+
+import uk.gov.justice.services.eventsourcing.repository.jdbc.event.Event;
+import uk.gov.justice.services.eventsourcing.repository.jdbc.eventstream.EventStream;
+import uk.gov.justice.services.eventsourcing.repository.jdbc.exception.InvalidSequenceIdException;
+import uk.gov.justice.services.messaging.JsonEnvelope;
+import uk.gov.justice.services.messaging.Metadata;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileFilter;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.file.Paths;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.time.ZonedDateTime;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Stream;
+
+import javax.sql.DataSource;
+
+import liquibase.Liquibase;
+import liquibase.database.jvm.JdbcConnection;
+import liquibase.resource.ClassLoaderResourceAccessor;
+import org.apache.commons.dbcp2.BasicDataSource;
+import org.apache.commons.io.filefilter.WildcardFileFilter;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+
+
+public class StreamTransformationIT {
+
+    private static final TestProperties TEST_PROPERTIES = new TestProperties("test.properties");
+
+    private static final UUID STREAM_ID = randomUUID();
+
+    private static TestEventLogJdbcRepository EVENT_LOG_JDBC_REPOSITORY;
+
+    private static TestEventStreamJdbcRepository EVENT_STREAM_JDBC_REPOSITORY;
+
+    private DataSource dataSource;
+
+    @Before
+    public void setUpDB() throws Exception {
+        dataSource = initEventStoreDb();
+        EVENT_LOG_JDBC_REPOSITORY = new TestEventLogJdbcRepository(dataSource);
+        EVENT_STREAM_JDBC_REPOSITORY = new TestEventStreamJdbcRepository(dataSource);
+    }
+
+    @Test
+    public void runReplayTool() throws Exception {
+        insertEventLogData();
+
+        runCommand(createCommandToExecuteStreamReplay());
+
+        assertTrue(eventStoreTransformedEventPresent());
+        assertTrue(originalEventStreamIsActive());
+        assertFalse(clonedStreamIsInactive());
+    }
+
+    private boolean clonedStreamIsInactive() {
+        final UUID clonedStreamId = EVENT_LOG_JDBC_REPOSITORY.findAll()
+                .filter(event -> event.getName().equals("system.events.cloned"))
+                .findFirst().get()
+                .getStreamId();
+
+        final boolean isActive = EVENT_STREAM_JDBC_REPOSITORY.findAll()
+                .filter(eventStream -> eventStream.getStreamId().equals(clonedStreamId))
+                .findFirst().get().isActive();
+
+        return isActive;
+    }
+
+    private boolean originalEventStreamIsActive() {
+        final Stream<EventStream> eventStream = EVENT_STREAM_JDBC_REPOSITORY.findAll();
```

Comment:
is this the same method as above? essentially isActive(streamId) just written in two ways. Maybe we can refactor this so it's a reusable method.